appscale/tools/local_state.py

```diff
@@ -365,61 +349,64 @@ def update_local_metadata(cls, options, db_master, head_node):
     """
     # find out every machine's IP address and what they're doing
     acc = AppControllerClient(head_node, cls.get_secret_key(options.keyname))
-    all_ips = [str(ip) for ip in acc.get_all_public_ips()]
     role_info = acc.get_role_info()
 
     infrastructure = options.infrastructure or 'xen'
 
     # write our yaml metadata file
-    yaml_contents = {
-      'load_balancer': str(head_node),
-      'table': options.table,
-      'secret': cls.get_secret_key(options.keyname),
-      'db_master': str(db_master),
-      'ips': all_ips,
-      'infrastructure': infrastructure,
-      'group': options.group,
+    appscalefile_contents = {
+      'infrastructure' : infrastructure,
+      'group' : options.group,
     }
 
     if infrastructure != "xen":
-      yaml_contents['zone'] = options.zone
+      appscalefile_contents['zone'] = options.zone
 
     if infrastructure == "gce":
-      yaml_contents['project'] = options.project
+      appscalefile_contents['project'] = options.project
 
     if infrastructure == 'azure':
-      yaml_contents['azure_subscription_id'] = options.azure_subscription_id
-      yaml_contents['azure_app_id'] = options.azure_app_id
-      yaml_contents['azure_app_secret_key'] = options.azure_app_secret_key
-      yaml_contents['azure_tenant_id'] = options.azure_tenant_id
-      yaml_contents['azure_resource_group'] = options.azure_resource_group
-      yaml_contents['azure_storage_account'] = options.azure_storage_account
-      yaml_contents['azure_group_tag'] = options.azure_group_tag
-
-    with open(cls.get_locations_yaml_location(options.keyname), 'w') \
-        as file_handle:
-      file_handle.write(yaml.dump(yaml_contents, default_flow_style=False))
+      appscalefile_contents['azure_subscription_id'] = options.azure_subscription_id
+      appscalefile_contents['azure_app_id'] = options.azure_app_id
+      appscalefile_contents['azure_app_secret_key'] = options.azure_app_secret_key
+      appscalefile_contents['azure_tenant_id'] = options.azure_tenant_id
+      appscalefile_contents['azure_resource_group'] = options.azure_resource_group
+      appscalefile_contents['azure_storage_account'] = options.azure_storage_account
+      appscalefile_contents['azure_group_tag'] = options.azure_group_tag
+
+    locations_json = {
+      'node_info': role_info,
+      'infrastructure_info': appscalefile_contents
+    }
 
     # and now we can write the json metadata file
     with open(cls.get_locations_json_location(options.keyname), 'w') \
         as file_handle:
-      file_handle.write(json.dumps(role_info))
+      file_handle.write(json.dumps(locations_json))
 
 
   @classmethod
-  def get_from_yaml(cls, keyname, tag):
+  def get_from_appscalefile(cls, tag, keyname):
     """Reads the YAML-encoded metadata on disk and returns the value associated
```

Comment:
Could you change the docs to specify that those tags are now got from the AppScalefile instead of the generic YAML.